From e22c887b55ec5e83db959c3b98d448947c886cc0 Mon Sep 17 00:00:00 2001
From: Pino Toscano <pino@kde.org>
Date: Sun, 14 Feb 2016 16:19:23 +0100
Subject: [PATCH] webinterface: rename major/minor to
 major_version/minor_version

Avoid conflicts with major/minor defines provided by GNU headers.
---
 plugins/webinterface/httpresponseheader.cpp | 10 +++++-----
 plugins/webinterface/httpresponseheader.h   |  6 +++---
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/plugins/webinterface/httpresponseheader.cpp b/plugins/webinterface/httpresponseheader.cpp
index 91d91fc..f7df1de 100644
--- a/plugins/webinterface/httpresponseheader.cpp
+++ b/plugins/webinterface/httpresponseheader.cpp
@@ -33,8 +33,8 @@ namespace kt
 		return QString::null;
 	}
 	
-	HttpResponseHeader::HttpResponseHeader(int response_code,int major,int minor) 
-		: response_code(response_code),major(major),minor(minor)
+	HttpResponseHeader::HttpResponseHeader(int response_code,int major_version,int minor_version) 
+		: response_code(response_code),major_version(major_version),minor_version(minor_version)
 	{
 	}
 	
@@ -42,8 +42,8 @@ namespace kt
 	{
 		response_code = hdr.response_code;
 		fields = hdr.fields;
-		major = hdr.major;
-		minor = hdr.minor;
+		major_version = hdr.major_version;
+		minor_version = hdr.minor_version;
 	}
 	
 	HttpResponseHeader::~HttpResponseHeader()
@@ -63,7 +63,7 @@ namespace kt
 	QString HttpResponseHeader::toString() const
 	{
 		QString str;
-		str += QString("HTTP/%1.%2 %3 %4\r\n").arg(major).arg(minor).arg(response_code).arg(ResponseCodeToString(response_code));
+		str += QString("HTTP/%1.%2 %3 %4\r\n").arg(major_version).arg(minor_version).arg(response_code).arg(ResponseCodeToString(response_code));
 		
 		QMap<QString,QString>::const_iterator itr = fields.begin();
 		while (itr != fields.end())
diff --git a/plugins/webinterface/httpresponseheader.h b/plugins/webinterface/httpresponseheader.h
index 7e0e1d0..a428d31 100644
--- a/plugins/webinterface/httpresponseheader.h
+++ b/plugins/webinterface/httpresponseheader.h
@@ -34,11 +34,11 @@ namespace kt
 	{
 		int response_code;
 		QMap<QString,QString> fields;
-		int major;
-		int minor;
+		int major_version;
+		int minor_version;
 		
 	public:
-		HttpResponseHeader(int response_code,int major = 1,int minor = 1);
+		HttpResponseHeader(int response_code,int major_version = 1,int minor_version = 1);
 		HttpResponseHeader(const HttpResponseHeader & hdr);
 		virtual ~HttpResponseHeader();
 		
-- 
2.7.0

